pkg/apiserver/handlers.go

```diff
@@ -72,13 +72,35 @@ func ReadOnly(handler http.Handler) http.Handler {
 	})
 }
 
+type LongRunningRequestCheck func(r *http.Request) bool
+
+// BasicLongRunningRequestCheck pathRegex operates against the url path, the queryParams match is case insensitive.
+// Any one match flags the request
+func BasicLongRunningRequestCheck(pathRegex *regexp.Regexp, queryParams map[string]string) LongRunningRequestCheck {
+	return func(r *http.Request) bool {
```

Comment:
> What prevents me from adding ?watch=true to all of my requests to avoid the rate limiter?

Nothing.  How much logic do we want to put in here to inspect the request?  It's possible to run a `RequestInfo` parse and make a more definitive determination.